pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -272,6 +278,14 @@ func validatePipelineTaskName(ctx context.Context, prefix string, i int, t Pipel
 		}
 		taskNames[t.Name] = struct{}{}
```

Comment:
The check for duplicate pipeline task names occurs inside this block when a taskref name is used.  So the validation hasn't been done if taskSpec is used, and now it also won't be done if a custom task has no taskRef name.  Should this duplicate name check be outside this block?  

I ran a test to confirm this.  If taskRef is used, a duplicate pipeline task name is caught here.  If taskSpec is used, a duplicate pipeline task name is not caught here, but it is caught by the reconciler when it builds the DAG:

```
Error from server (BadRequest): error when creating "C:\\dbg\\tekton\\samples\\paralleltasks.yaml": admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: invalid value: task task1 is already present in Graph, can't add it again: duplicate pipeline task: spec.tasks
```

Reply:
Good catch! Fixed, and updated the test so it would have caught this.